semiosis/contexts/mock_context.py

```
@@ -0,0 +1,132 @@
+        return elements
+    
+    def extract_context(self) -> List[ContextElement]:
+        """
+        Extract all available context elements from the system.
+        
+        Returns:
+            List of ContextElement objects
+        """
+        return self.context_elements
+    
+    def filter_context(self, query: str, max_elements: int = None) -> List[ContextElement]:
```

Ruff (0.13.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)